.github/workflows/ci.yml

```diff
@@ -0,0 +1,145 @@
+name: aiida-core
+
+on: [push, pull_request]
+
+jobs:
+
+  conda:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Conda install
+      env:
+        PYTHON_VERSION: 3.7
+      run:
+        .github/workflows/conda.sh
+
+  docs:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Install system dependencies
+      run: |
+        sudo apt update
+        sudo apt install texlive-base texlive-generic-recommended texlive-fonts-recommended texlive-latex-base texlive-latex-recommended texlive-latex-extra dvipng dvidvi
+
+    - name: Install python dependencies
+      run:
+        pip install -e .[docs,testing]
+
+    - name: Build documentation
+      env:
+        READTHEDOCS: 'True'
+      run:
+        SPHINXOPTS='-nW' make -C docs html
+
+  pre-commit:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Install system dependencies
+      run: |
+        sudo apt update
+        sudo apt install libkrb5-dev ruby ruby-dev
+
+    - name: Install python dependencies
+      run:
+        pip install -e .[all]
+
+    - name: Run pre-commit
+      run:
+        pre-commit run --all-files || ( git status --short ; git diff ; exit 1 )
+
+  tests:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        backend: ['django', 'sqlalchemy']
```

Comment:
if the number of concurrent jobs is indeed not a problem, I would suggest we test at least django also on python 3.5

Reply:
I have added python 3.6 to the matrix strategy for now. Is there a reason you prefer 3.5 over 3.6? We officially support both though but a matrix of 6 is maybe overdoing it a bit?

Reply:
The thinking was that if it is working on 3.5 and 3.7 it's likely to work on 3.6 as well.
And I think it probably suffices to test sqlalchemy on one python version... anyhow up to you.

Reply:
OK, that makes sense. I will switch it to python 3.5